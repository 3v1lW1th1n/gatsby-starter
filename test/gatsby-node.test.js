/* eslint-disable no-undef */
jest.mock('path');
const {createPages, onNodeCreate} = require('../gatsby-node');

describe('gatsby-node', () => {
  const boundActionCreators = {};

  describe('createPages', () => {
    let graphql;

    beforeEach(() => {
      boundActionCreators.upsertPage = jest.fn();
      graphql = jest.fn();
    });

    it('should create blog posts', () => {
      graphql.mockReturnValueOnce(
        Promise.resolve(
          {
            data: {
              posts: {
                edges: [
                  {
                    node: {
                      slug: '/blog/2017-04-18--welcoming/'
                    }
                  },
                  {
                    node: {
                      slug: '/blog/2017-05-02--article-2/'
                    }
                  },
                  {
                    node: {
                      slug: '/blog/2017-05-02--article-1/'
                    }
                  },
                  {
                    node: {
                      slug: '/docs/dont-take-me/'
                    }
                  }
                ]
              }
            }
          }
        )
      );

      return createPages({graphql, boundActionCreators})
        .then(() => {
          expect(boundActionCreators.upsertPage.mock.calls).toMatchSnapshot();
        });
    });
  });

  describe('onNodeCreate', () => {
    let getNode;

    beforeEach(() => {
      boundActionCreators.updateNode = jest.fn();
      getNode = jest.fn();
    });

    it('should create slugs for MarkdownRemark file', () => {
      getNode.mockReturnValue(
        {
          relativePath: 'blog/2017-04-18--welcoming/index.md'
        }
      );
      const node = {
        type: 'MarkdownRemark',
        parent: 'parent'
      };
      onNodeCreate({node, boundActionCreators, getNode});

      expect(boundActionCreators.updateNode.mock.calls).toMatchSnapshot();
    });

    it('should do nothing on unknown type', () => {
      getNode.mockReturnValue(
        {
          relativePath: 'blog/2017-04-18--welcoming/index.md'
        }
      );
      const node = {
        type: 'unknown',
        parent: 'parent'
      };
      onNodeCreate({node, boundActionCreators, getNode});

      expect(boundActionCreators.updateNode.mock.calls.length).toBe(0);
    });
  });
});
