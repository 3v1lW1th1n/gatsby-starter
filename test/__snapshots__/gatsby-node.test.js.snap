// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-node createPages should create blog posts 1`] = `
Array [
  Array [
    Object {
      "component": "base-path/src/templates/blog-post.tsx",
      "context": Object {
        "slug": "/blog/2017-04-18--welcoming/",
      },
      "path": "/blog/2017-04-18--welcoming/",
    },
  ],
  Array [
    Object {
      "component": "base-path/src/templates/blog-post.tsx",
      "context": Object {
        "slug": "/blog/2017-05-02--article-2/",
      },
      "path": "/blog/2017-05-02--article-2/",
    },
  ],
  Array [
    Object {
      "component": "base-path/src/templates/blog-post.tsx",
      "context": Object {
        "slug": "/blog/2017-05-02--article-1/",
      },
      "path": "/blog/2017-05-02--article-1/",
    },
  ],
]
`;

exports[`gatsby-node createPages should create tags pages 1`] = `
Array [
  Array [
    Object {
      "component": "base-path/src/templates/blog-post.tsx",
      "context": Object {
        "slug": "/blog/2017-04-18--welcoming/",
      },
      "path": "/blog/2017-04-18--welcoming/",
    },
  ],
  Array [
    Object {
      "component": "base-path/src/templates/blog-post.tsx",
      "context": Object {
        "slug": "/blog/2017-05-02--article-2/",
      },
      "path": "/blog/2017-05-02--article-2/",
    },
  ],
  Array [
    Object {
      "component": "base-path/src/templates/blog-post.tsx",
      "context": Object {
        "slug": "/blog/2017-05-02--article-1/",
      },
      "path": "/blog/2017-05-02--article-1/",
    },
  ],
  Array [
    Object {
      "component": "base-path/src/templates/blog-post.tsx",
      "context": Object {
        "slug": "/blog/2017-05-02--article-0/",
      },
      "path": "/blog/2017-05-02--article-0/",
    },
  ],
  Array [
    Object {
      "component": "base-path/src/templates/tags-page.tsx",
      "context": Object {
        "tag": "starter",
      },
      "path": "/blog/tags/starter/",
    },
  ],
  Array [
    Object {
      "component": "base-path/src/templates/tags-page.tsx",
      "context": Object {
        "tag": "gatsby",
      },
      "path": "/blog/tags/gatsby/",
    },
  ],
  Array [
    Object {
      "component": "base-path/src/templates/tags-page.tsx",
      "context": Object {
        "tag": "test",
      },
      "path": "/blog/tags/test/",
    },
  ],
]
`;

exports[`gatsby-node onNodeCreate should create slugs for MarkdownRemark file 1`] = `
Array [
  Array [
    Object {
      "parent": "parent",
      "slug": "/blog/2017-04-18--welcoming/",
      "type": "MarkdownRemark",
    },
  ],
]
`;
